File: packages/python/dirsql/resolve_entries.py
"""Resolution of one config's ``[[dirsql.extension]]`` entries to literal paths."""

from __future__ import annotations

from .resolve_extension import resolve_extension_path


def _resolve_entries(entries, base):
    specs = []
    for e in entries:
        entrypoint = e.get("entrypoint")
        specs.append(
            {
                "path": resolve_extension_path(
                    e["path"], base=base, resolve_relative=True
                ),
                "entrypoint": entrypoint if isinstance(entrypoint, str) else None,
            }
        )
    return specs

File: packages/ts/src/resolve-entries.ts
import type { ExtensionSpec } from "./dirsql.js";
import type { Toml } from "./load-extension-entries.js";
import { resolveExtensionPath } from "./resolve-extension.js";

/** Resolve a config's entries to specs, against that config's own directory. */
export function resolveEntries(entries: Toml[], base: string): ExtensionSpec[] {
  return entries.map((e) => ({
    path: resolveExtensionPath(e.path as string, base, true),
    entrypoint: typeof e.entrypoint === "string" ? e.entrypoint : undefined,
  }));
}
